### ISSUE TITLE
Bug in HDPrivateKey._getDerivationIndexes

### ISSUE THREAD

**Issue by fanatid** [bitpay/bitcore-lib#46]
```js
var bitcore = require('bitcore-lib')
console.log(bitcore.HDPrivateKey._getDerivationIndexes('m/2.1/3')) // [ 2.1, 3 ]
```

Instead `[ 2.1, 3 ]` result should be `null`

Discussion in bitcoinjs-lib: https://github.com/bitcoinjs/bitcoinjs-lib/pull/538

**Comment by braydonf** [comment]
Shouldn't it throw with an invalid argument?

**Comment by fanatid** [comment]
No. If `_getDerivationIndexes` return `null`, `_deriveFromString` throw error on checking path: https://github.com/bitpay/bitcore-lib/blob/4430479cea370ea07ed02272776932f76bcd2908/lib/hdprivatekey.js#L211

**Comment by braydonf** [comment]
It looks like the behavior currently is that `hdPrivateKey.derive('m/2/3')` and `hdPrivateKey.derive('m/2.1/3')` will generate the same key.
